src/WindowsAzureMessaging/WindowsAzureMessaging/Helpers/MSInstallationManager.m

```diff
@@ -140,4 +140,15 @@ + (NSURL *)fixupEndpoint:(NSURL *)endPoint scheme:(NSString *)scheme {
     return [NSURL URLWithString:modifiedEndpoint];
 }
 
++ (NSString *)getApiVersionFromEndpoint:(NSString *)endpoint {
+    if ([endpoint containsString:AzureEnvironmentINT7] || [endpoint containsString:AzureEnvironmentPROD]) {
+        return ApiVersion2020_06;
+    } else if ([endpoint containsString:AzureEnvironmentFFPROD] || [endpoint containsString:AzureEnvironmentBFPROD] ||
+               [endpoint containsString:AzureEnvironmentCHPROD]) {
+        return ApiVersion2017_04;
```

Comment:
Same as in Java SDK, I think we can remove all conditional for private clouds and just replace it with "else".
And I still don't like mentioning of all private clouds :)